### PR TITLE
Support update and switchover DR

### PR DESCRIPTION
**Pull request by posriniv**
```
./ybm cluster dr update --dr-name sample-config --cluster-name alluring-goat --databases test1,test2
./ybm cluster dr switchover --cluster-name alluring-goat --dr-name sample-config
```

### CODE REVIEW

File: cmd/cluster/dr/switchover_dr.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	Long:  `Switchover DR for a cluster`,
+	Run: func(cmd *cobra.Command, args []string) {
+		authApi, err := ybmAuthClient.NewAuthApiClient()
+		if err != nil {
+			logrus.Fatalf(ybmAuthClient.GetApiErrorDetails(err))
+		}
+		authApi.GetInfo("", "")
+
+		drName, _ := cmd.Flags().GetString("dr-name")
+		clusterId, err := authApi.GetClusterIdByName(ClusterName)
```
Comment (artem-mindrov) [inline cmd/cluster/dr/switchover_dr.go:42]: What's `ClusterName` here? Where does it come from?

File: cmd/cluster/dr/update_dr.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+		for _, namespace := range namespacesResp.Data {
+			dbNameToIdMap[namespace.GetName()] = namespace.GetId()
+		}
+		databaseIds := []string{}
+		for _, databaseString := range databases {
+			for _, database := range strings.Split(databaseString, ",") {
+				if databaseId, exists := dbNameToIdMap[database]; exists {
+					databaseIds = append(databaseIds, databaseId)
+				} else {
+					logrus.Fatalf("The database %s doesn't exist", database)
```
Comment (artem-mindrov) [inline cmd/cluster/dr/update_dr.go:67]: IIUC we're effectively duplicating the API level checks here, do we need this?

File: cmd/cluster/dr/update_dr.go [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+		if viper.GetBool("wait") {
+			returnStatus, err := authApi.WaitForTaskCompletion(clusterId, ybmclient.ENTITYTYPEENUM_CLUSTER, ybmclient.TASKTYPEENUM_EDIT_DR, []string{"FAILED", "SUCCEEDED"}, msg)
+			if err != nil {
+				logrus.Fatalf("error when getting task status: %s", err)
+			}
+			if returnStatus != "SUCCEEDED" {
+				logrus.Fatalf("Operation failed with error: %s", returnStatus)
+			}
+			fmt.Printf("The DR %s has been updated\n", formatter.Colorize(drName, formatter.GREEN_COLOR))
```
Comment (artem-mindrov) [inline cmd/cluster/dr/update_dr.go:89]:
```suggestion
			fmt.Printf("DR config %s has been updated\n", formatter.Colorize(drName, formatter.GREEN_COLOR))
```

File: cmd/cluster/dr/switchover_dr.go [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+		if viper.GetBool("wait") {
+			returnStatus, err := authApi.WaitForTaskCompletion(clusterId, ybmclient.ENTITYTYPEENUM_CLUSTER, ybmclient.TASKTYPEENUM_DR_SWITCHOVER, []string{"FAILED", "SUCCEEDED"}, msg)
+			if err != nil {
+				logrus.Fatalf("error when getting task status: %s", err)
+			}
+			if returnStatus != "SUCCEEDED" {
+				logrus.Fatalf("Operation failed with error: %s", returnStatus)
+			}
+			fmt.Printf("The switchover for DR %s is successful\n", formatter.Colorize(drName, formatter.GREEN_COLOR))
```
Comment (artem-mindrov) [inline cmd/cluster/dr/switchover_dr.go:67]:
```suggestion
			fmt.Printf("Switchover for DR config %s is successful\n", formatter.Colorize(drName, formatter.GREEN_COLOR))
```

File: cmd/cluster/dr/update_dr.go (hunk 2 above)
Comment (posriniv) [reply cmd/cluster/dr/update_dr.go:67]: Yes, we would like to validate and fail in the CLI itself.

File: cmd/cluster/dr/switchover_dr.go (hunk 1 above)
Comment (artem-mindrov) [reply cmd/cluster/dr/switchover_dr.go:42]: NVM I see it's in the args. Do we need it there though? Isn't DR config name enough?

Comment (posriniv) [reply cmd/cluster/dr/switchover_dr.go:42]: It comes from `--cluster-name` that we provide in the command line.

File: cmd/cluster/dr/update_dr.go (hunk 2 above)
Comment (artem-mindrov) [reply cmd/cluster/dr/update_dr.go:67]: But why the need for this additional validation? The API call will fail synchronously anyway

Comment (posriniv) [reply cmd/cluster/dr/update_dr.go:67]: Well, actually, we need the above API call to fetch database IDs from names, hence it is not another wasted API call, we are just checking an in-memory map.

Comment (artem-mindrov) [reply cmd/cluster/dr/update_dr.go:67]: Ah it's translation code, my bad

File: cmd/cluster/dr/switchover_dr.go (hunk 1 above)
Comment (artem-mindrov) [reply cmd/cluster/dr/switchover_dr.go:42]: Yes I see but isn't cluster name redundant still?

Comment (posriniv) [reply cmd/cluster/dr/switchover_dr.go:42]: I have designed the DR config to be a sub resource of the cluster resource, hence the cluster name needs to be given similar to the API.

File: cmd/cluster/dr/update_dr.go [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+		}
+
+		updateDrRequest := ybmclient.NewEditXClusterDrRequest(*ybmclient.NewEditXClusterDrSpec(databaseIds))
+		drResp, response, err := authApi.EditXClusterDr(clusterId, drId).EditXClusterDrRequest(*updateDrRequest).Execute()
+		if err != nil {
+			logrus.Debugf("Full HTTP response: %v", response)
+			logrus.Fatalf(ybmAuthClient.GetApiErrorDetails(err))
+		}
+
+		msg := fmt.Sprintf("The DR %s is being updated", formatter.Colorize(drName, formatter.GREEN_COLOR))
```
Comment (daniel-yb) [inline cmd/cluster/dr/update_dr.go:79]:
```suggestion
		msg := fmt.Sprintf("DR config %s is being updated", formatter.Colorize(drName, formatter.GREEN_COLOR))
```

**Review by daniel-yb — APPROVED** [review]

**Pull request merged**